packages/itwin/tree-widget/src/test/trees/models-tree/stateless/ModelsTreeFiltering.test.ts

```diff
@@ -910,14 +967,290 @@ describe("Models tree", () => {
+          const rootElement = insertPhysicalElement({ builder, userLabel: `root element`, modelId: model.id, categoryId: category.id });
+          const physicalElement1 = insertPhysicalElement({
+            builder,
+            userLabel: `element 1`,
+            modelId: model.id,
+            categoryId: category.id,
+            parentId: rootElement.id,
+          });
+          const physicalElement2 = insertPhysicalElement({
+            builder,
+            userLabel: `element 2`,
+            modelId: model.id,
+            categoryId: category.id,
+            parentId: rootElement.id,
+          });
+          const testElement1 = insertPhysicalElement({
+            builder,
+            classFullName: testSchema.items.SubModelablePhysicalObject.fullName,
+            userLabel: `test element 1`,
+            modelId: model.id,
+            categoryId: category.id,
+            parentId: rootElement.id,
+          });
+          const testElement2 = insertPhysicalElement({
+            builder,
+            classFullName: testSchema.items.SubModelablePhysicalObject.fullName,
+            userLabel: `test element 2`,
+            modelId: model.id,
+            categoryId: category.id,
+            parentId: rootElement.id,
+          });
+          return { rootSubject, model, category, rootElement, physicalElement1, physicalElement2, testElement1, testElement2 };
+        },
+        (x) => [
+          [x.rootSubject, x.model, x.category, x.rootElement, x.testElement1],
+          [x.rootSubject, x.model, x.category, x.rootElement, x.testElement2],
+        ],
+        (x) => [{ parentKey: x.rootElement, parentType: "element", classes: [x.testElement1.className] }],
+        undefined,
+        (x) => [
+          NodeValidators.createForInstanceNode({
+            instanceKeys: [x.rootSubject],
+            autoExpand: true,
+            children: [
+              NodeValidators.createForInstanceNode({
+                instanceKeys: [x.model],
+                label: "model-x",
+                autoExpand: true,
+                children: [
+                  NodeValidators.createForInstanceNode({
+                    instanceKeys: [x.category],
+                    label: "category-x",
+                    autoExpand: true,
+                    children: [
+                      NodeValidators.createForClassGroupingNode({
+                        label: "Physical Object",
+                        autoExpand: true,
+                        children: [
+                          NodeValidators.createForInstanceNode({
+                            instanceKeys: [x.rootElement],
+                            label: /^root element/,
+                            children: [
+                              NodeValidators.createForClassGroupingNode({
+                                label: "Test Physical Object",
+                                autoExpand: true,
```

Comment:
Ideally, the hierarchy should be auto-expanded up to the grouping node, but the grouping node itself should not. Can we do that?

Reply:
Filed a separate issue for tracking this: https://github.com/iTwin/viewer-components-react/issues/947